app/assets/javascripts/sitewide/tiny_mce_file_upload_plugin.js

```diff
@@ -1,308 +1,81 @@
 /* eslint no-underscore-dangle: "off" */
 /* eslint no-use-before-define: "off" */
 /* eslint no-restricted-syntax: ["off", "BinaryExpression[operator='in']"] */
-/* global tinymce I18n GLOBAL_CONSTANTS */
+/* global tinymce I18n GLOBAL_CONSTANTS HelperModule*/
 (function() {
   'use strict';
 
   tinymce.PluginManager.requireLangPack('customimageuploader');
 
   tinymce.create('tinymce.plugins.CustomImageUploader', {
-    CustomImageUploader: function(ed, url) {
-      var form;
+    CustomImageUploader: function(ed) {
       var iframe;
-      var win;
-      var throbber;
       var editor = ed;
       var textAreaElement = $('#' + ed.id);
-      function showDialog() {
-        var ctrl;
-        var cycle;
-        var el;
-        var body;
-        var containers;
-        var inputs;
-        win = editor.windowManager.open({
-          title: I18n.t('tiny_mce.upload_window_title'),
-          width: 520 + parseInt(editor.getLang(
-            'customimageuploader.delta_width', 0
-          ), 10),
-          height: 180 + parseInt(editor.getLang(
-            'customimageuploader.delta_height', 0
-          ), 10),
-          body: [
-            { type: 'iframe', url: 'javascript:void(0)' },
-            {
-              type: 'textbox',
-              name: 'file',
-              classes: 'image-loader',
-              label: I18n.t('tiny_mce.upload_window_label'),
-              subtype: 'file'
-            },
-            {
-              type: 'container',
-              classes: 'error',
-              html: "<p style='color: #b94a48;'>&nbsp;</p>"
+
+      function loadFiles() {
+        let $fileInput;
+        $('#tinymce_current_upload').remove();
+        $(editor.container).prepend('<input type="file" multiple id="tinymce_current_upload" style="opacity: 0; height: 0">');
+        $fileInput = $('#tinymce_current_upload');
+        $fileInput.trigger('click');
+
+        $fileInput.change(function() {
+          let formData = new FormData();
+          let files = $('#tinymce_current_upload')[0].files;
+
+          Array.from(files).forEach(file => formData.append('files[]', file, file.name));
+
+          $.ajax({
+            url: '/tiny_mce_assets',
+            type: 'POST',
+            data: formData,
+            processData: false,
+            contentType: false,
+            beforeSend: function(xhr) {
```

Comment:
Maybe create some sitewide method to validate file size?

Reply:
Hm.. where we can actually reuse it? We are checking array of files (file input, fileList) only here I think..?

Reply:
Steps, results, inventories. It will be a general method to check the file size. Because of our file size limit general.

Reply:
Yes, but what we can reuse is only that if check. The method here will be different from others, where we have direct upload...

Reply:
It will be the same, you just check files size.